linkerd/app/outbound/src/lib.rs

```diff
@@ -427,13 +428,19 @@ impl<A: OrigDstAddr> Config<A> {
                 http_server.into_inner(),
                 h2_settings,
                 drain.clone(),
-                disable_protocol_detection_for_ports.clone(),
             );
 
+            let svc = svc::stack(tcp_server)
+                .push(DetectProtocolLayer::new(ProtocolDetect::new(
+                    disable_protocol_detection_for_ports.clone(),
+                )))
+                .push_timeout(detect_protocol_timeout)
+                .check_service_response::<(tls::accept::Meta, BoxedIo), ()>();
+
             // The local application does not establish mTLS with the proxy.
             let no_tls: tls::Conditional<identity::Local> =
                 Conditional::None(tls::ReasonForNoPeerName::Loopback.into());
-            let accept = tls::AcceptTls::new(no_tls, tcp_server)
+            let accept = tls::AcceptTls::new(no_tls, svc.into_inner())
```

Comment:
Don't we need to timeout outside of the TLS detection? Or, why don't we encounter this problem with TLS detection?

Reply:
Errr, this won't matter on the outbound side. But it will on the inbound side.